server/app/service/ScalaDynamicNoTraitRunner.scala

```diff
@@ -1,14 +1,30 @@
 package service
 
-import scala.util.Try
+import monifu.concurrent.Scheduler
+import monifu.reactive.Observable
+import monifu.reactive.OverflowStrategy.DropOld
+import monifu.reactive.channels.PublishChannel
+import shared.Line
 
-trait ScalaDynamicNoTraitRunner extends DynamicExecution with TryBlock {
+import scala.concurrent.Future
+import scala.util.control.NonFatal
+
+trait ScalaDynamicNoTraitRunner extends DynamicExecution {
   /**
    * Runs dynamic solution as well as dynamic suite using the structural type for test, instead of explicitly defined
    * trait
    */
-  def execSuiteNoTrait(solution: String, suite: String): Try[String] = {
-    val patchedSolution = classDefPattern.replaceFirstIn(solution, s"class $userClass ")
-    executeDynamic(suite, patchedSolution)
+  def execSuiteNoTrait(solution: String, suite: String)
```

Comment:
I do not like we again mixing scalatest runner with something else like monifu in this case. Scala test runner just runs tests. Is it possible to extract this channel stuff outside of this class?